crates/driver/src/domain/competition/mod.rs

```diff
@@ -103,19 +103,44 @@ impl Competition {
             }
         });
 
+        // Merge solutions
+        let mut merged: Vec<Solution> = Vec::new();
+        for solution in solutions {
+            let mut extension = vec![];
+            for already_merged in merged.iter() {
+                match solution.merge(already_merged) {
+                    Ok(merged) => {
+                        observe::merged(&solution, already_merged, &merged);
+                        extension.push(merged);
+                        continue;
+                    }
+                    Err(err) => {
+                        observe::not_merged(&solution, already_merged, err);
+                    }
+                }
+            }
+            // At least insert the current solution
+            extension.push(solution);
+            merged.extend(extension);
+        }
+        // Sort by "simplest", ie least merged solution.
+        // Maybe we should sort by score instead?
```

Comment:
@harisang brought up at some point that we have no way of knowing the score of a merged solution so using that for determining a sort order is probably not wise.
Prioritizing solutions that are the closest to what the solver originally intended seems like the most reasonable choice here.
But since we now have reth nodes we might not have to sweat it anyway. ;)

Reply:
The reason we only knew the score for settlements (not solutions) was that they required simulation to get the gas costs. Thanks to CiP38 (rank by surplus) this goes away and so we can now actually compute the score from the solution (cf. #2448).

So I think we could still debate if we want to start with the highest scored solution (even if it's more risky to fail because it'll likely consist of many merged settlements.